torch/csrc/jit/pickler.h

```diff
@@ -114,21 +114,27 @@ struct WriteableTensorData {
   size_t numel() const {
     return tensor_.storage().numel();
   }
+  const std::string& key() const {
+    return key_;
+  }
+  const at::Tensor& tensor() const {
```

Comment:
This should not expose the tensor. The place where it is being used illustrates a location where having tensor references rather than raw data _would_ be appropriate. It is possible more design discussion needs to occur to clarify requirements for the three modes the pickler is used (torch.save compat, network communication, torch.jit.save)